daemon/containerd/image_list.go

```diff
@@ -154,9 +154,13 @@ func (i *ImageService) Images(ctx context.Context, opts imagetypes.ListOptions)
 
 	for _, img := range uniqueImages {
 		image, allChainsIDs, err := i.imageSummary(ctx, img, platformMatcher, opts, tagsByDigest)
-		if err != nil || image == nil {
+		if err != nil {
 			return nil, err
 		}
+		// No error, but image should be skipped.
```

Comment:
Not for this PR, but should we consider imageSummary to return a sentinel (errdefs.NotFound) error?

Reply:
Ah,  right, so we have some fun case here; https://github.com/moby/moby/blob/979f03f9f6c1f13d6c994f284dc1b8297ce456fd/daemon/containerd/image_list.go#L284-L290

And https://github.com/moby/moby/blob/979f03f9f6c1f13d6c994f284dc1b8297ce456fd/daemon/containerd/image_list.go#L188

Yeah, looks like we may need to have a look at those.

Reply:
Looks like we currently fail to pull those (but same without containerd snapshotter);

```bash
docker pull tianon/scratch:index
Error response from daemon: no match for platform in manifest: not found

docker pull tianon/scratch:list
Error response from daemon: no match for platform in manifest: not found
```

I was curious if it would pick `linux/amd64` (as that was/is default for "no platform");

```bash
docker pull --platform=linux/amd64 tianon/scratch:list
Error response from daemon: no match for platform in manifest: not found

docker pull --platform=linux/amd64 tianon/scratch:index
Error response from daemon: no match for platform in manifest: not found
```